crates/cairo-lint-core/src/lints/enum_variant_names.rs

```diff
@@ -0,0 +1,178 @@
+use crate::context::{CairoLintKind, Lint};
+use cairo_lang_defs::ids::{LanguageElementId, ModuleItemId};
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_semantic::db::SemanticGroup;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{
+    ast::ItemEnum as AstEnumItem, SyntaxNode, Terminal, TypedSyntaxNode,
+};
+
+pub struct EnumVariantNames;
+
+/// ## What it does
+///
+/// Detects enumeration variants that are prefixed or suffixed by the same characters.
+///
+/// ## Example
+///
+/// ```cairo
+/// enum Cake {
+///     BlackForestCake,
+///     HummingbirdCake,
+///     BattenbergCake,
+/// }
+/// ```
+///
+/// Can be simplified to:
+///
+/// ```cairo
+/// enum Cake {
+///     BlackForest,
+///     Hummingbird,
+///     Battenberg,
+/// }
+/// ```
+impl Lint for EnumVariantNames {
+    fn allowed_name(&self) -> &'static str {
+        "enum_variant_names"
+    }
+
+    fn diagnostic_message(&self) -> &'static str {
+        "All enum variants are prefixed or suffixed by the same characters."
+    }
+
+    fn kind(&self) -> CairoLintKind {
+        CairoLintKind::EnumVariantNames
+    }
+
+    fn has_fixer(&self) -> bool {
+        true
+    }
+
+    fn fix(&self, db: &dyn SyntaxGroup, node: SyntaxNode) -> Option<(SyntaxNode, String)> {
+        fix_enum_variant_names(db, node)
+    }
+}
+
+pub fn check_enum_variant_names(
+    db: &dyn SemanticGroup,
+    item: &ModuleItemId,
+    diagnostics: &mut Vec<PluginDiagnostic>,
+) {
+    if let ModuleItemId::Enum(enum_id) = item {
+        if let Ok(variants) = db.enum_variants(*enum_id) {
+            let variant_names: Vec<String> = variants.iter().map(|v| v.0.to_string()).collect();
+
+            let (prefix, suffix) = get_prefix_and_suffix(&variant_names);
+
+            if !prefix.is_empty() || !suffix.is_empty() {
+                diagnostics.push(PluginDiagnostic {
+                    stable_ptr: enum_id.untyped_stable_ptr(db.upcast()),
+                    message: EnumVariantNames.diagnostic_message().to_string(),
+                    severity: Severity::Warning,
+                });
+            }
+        }
+    }
+}
```

Comment:
total nitpick for educational purposes, this would be preferable here:

```suggestion
    let ModuleItemId::Enum(enum_id) = item else { return };
    let Ok(variants) = db.enum_variants(*enum_id) else { return };
    let variant_names: Vec<String> = variants.iter().map(|v| v.0.to_string()).collect();

    let (prefix, suffix) = get_prefix_and_suffix(&variant_names);

    if !prefix.is_empty() || !suffix.is_empty() {
        diagnostics.push(PluginDiagnostic {
            stable_ptr: enum_id.untyped_stable_ptr(db.upcast()),
            message: EnumVariantNames.diagnostic_message().to_string(),
            severity: Severity::Warning,
        });
    }
}
```